app/src/main/java/no/usn/mob3000/domain/viewmodel/socials/ProfileViewModel.kt

```diff
@@ -0,0 +1,296 @@
+    val currentUserProfile: StateFlow<UserProfile?> = _currentUserProfile
+
+    private val _userGameStats = MutableStateFlow<Result<UserGameStats>>(Result.success(UserGameStats(0, 0, 0)))
+    val userGameStats: StateFlow<Result<UserGameStats>> = _userGameStats
+
+    /* ID/Selection */
+    private val _selectedUser = mutableStateOf<UserProfile?>(null)
+    val selectedUser: State<UserProfile?> = _selectedUser
+
+    private val _userId = MutableStateFlow<String?>(null)
+    val userId: StateFlow<String?> = _userId
+
+    /* Friend requests. */
+    private val _friends = MutableStateFlow<Result<List<FriendData>>>(Result.success(emptyList()))
+    val friends: StateFlow<Result<List<FriendData>>> = _friends
+
+    private val _nonFriends = MutableStateFlow<Result<List<UserProfile>>>(Result.success(emptyList()))
+    val nonFriends: StateFlow<Result<List<UserProfile>>> = _nonFriends
+
+    private val _friendRequests = MutableStateFlow<Result<List<FriendRequestData>>>(Result.success(emptyList()))
+    val friendRequests: StateFlow<Result<List<FriendRequestData>>> = _friendRequests
+
+    /**
+     * Initializes the ViewModel by fetching the current user's ID and fetching their profile. Used primary to ensure better refreshes, as this part of the system will not
+     * have access to the room database.
+     */
+    init {
+        viewModelScope.launch {
+            _userId.value = getCurrentUserIdUseCase.getCurrentUserId()
+            userId.value?.let { userId ->
+                fetchUser(userId)
+                fetchUserGameStats()
+            }
+        }
+    }
+
+    /***************Friend request***************/
+    /**
+     * Inserts a friend request into the database.
+     *
+     * @param toUser The ID of the user to send the friend request to.
+     */
+    fun insertFriendRequest(
+        toUser: String
+    ) {
+        viewModelScope.launch {
+            friendRequestUseCase.insert(toUser)
+        }
+    }
+    /**
+     *
+     *
+     * Accepts a friend request.
+     */
+    fun acceptFriendRequest(friendRequestId: String) {
+        viewModelScope.launch {
+            val result = friendRequestUseCase.accept(friendRequestId)
+            result.onSuccess {
+                fetchFriendRequests()
+                fetchFriends()
+            }.onFailure {
+                Log.e("ProfileViewModel", "Error when accepting: ${it.message}")
+            }
+        }
+    }
+    /**
+     *
+     *
+     * Declines a friend request.
+     */
+    fun declineFriendRequest(friendRequestId: String) {
+        viewModelScope.launch {
+            val result = friendRequestUseCase.decline(friendRequestId)
+            result.onSuccess {
+                fetchFriendRequests()
+            }.onFailure {
+                Log.e("ProfileViewModel", "Error when declining: ${it.message}")
+            }
+        }
+    }
+    /***************User profile***************/
+
+
+    /**

```

Comment:
Unnecessary double empty lines. It's fine to have an empty line between this and the "section" KDoc separator, however, that's not necessary and is not what [this comment refers to](https://github.com/frigvid/mob3000-group11/pull/25/files#r1848056317).